commands/service_publish.go

```diff
@@ -0,0 +1,105 @@
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/docker/docker/pkg/archive"
+	"github.com/mesg-foundation/core/ipfs"
+	"github.com/mesg-foundation/core/service/importer"
+	"github.com/spf13/cobra"
+)
+
+type servicePublishCmd struct {
+	baseCmd
+
+	path string
+	ipfs *ipfs.IPFS
+
+	e ServiceExecutor
+}
+
+type marketplaceData struct {
+	Version int `json:"version,omitempty"`
+	Service struct {
+		Deployment struct {
+			Source string `json:"source,omitempty"`
```

Comment:
i think, we can reduce _omitempty_'s a bit. it doesn't have any effect where a field is always set(required), for example in this case(service, deployment and source fields), definition and version(?) fields.